ui-v3/src/components/dashboard/RecentActivity.tsx

```diff
@@ -1,59 +1,87 @@
-
 import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
 import { Button } from "@/components/ui/button";
-import { Activity, ArrowUpRight, ArrowDownLeft, TrendingUp } from "lucide-react";
+import { Activity, ArrowUpRight, ArrowDownLeft, TrendingUp, Loader2 } from "lucide-react";
 import { Link, useParams } from "react-router-dom";
 import SecondaryButton from "../ui/secondary-button";
-
-interface ActivityItem {
-  id: string;
-  type: 'send' | 'receive' | 'stacking';
-  asset: string;
-  amount: string;
-  timestamp: string;
-  status: 'confirmed' | 'pending' | 'failed';
-}
+import { useCallback, useEffect, useState } from "react";
+import { Transaction, TransactionDataService } from "@/services/transactionDataService";
+import { formatAmount } from "@/lib/txFormatUtils";
+import { fetchStxUsdPrice } from "@/lib/stxPrice";
+import { Skeleton } from "@/components/ui/skeleton";
 
 interface RecentActivityProps {
-  activities?: ActivityItem[];
+  walletAddress?: string; // connected wallet
+  smartWalletAddress?: string; // smart wallet
 }
 
-const RecentActivity = ({ activities = [] }: RecentActivityProps) => {
-  const { walletId } = useParams();
+const transactionService = new TransactionDataService();
```

Comment:
you can use `useBlockchainServices` to retrieve the transactionDataService. That hook is setup to handle the demo mode.